src/connectors/infinite-hits/connectInfiniteHits.ts

```diff
@@ -207,18 +203,20 @@ const connectInfiniteHits: InfiniteHitsConnector = function connectInfiniteHits(
           widgetType: this.$$type!,
         });
 
+        const cachedHits = cache.read({ state: helper.state }) || {};
+        const pageNumbers = Object.keys(cachedHits).map(Number);
+        firstReceivedPage = Math.min(...pageNumbers);
```

Comment:
moving this to a global will be harder for getWidgetRenderState I think, not sure what the alternative is though.

Could it be stored in the cache too maybe? (note that we would need to have some kind of migration or something like that for active caches, although that won't be an issue, since we use sessionStorage)

Reply:
we should put in the release notes of the version containing this that if you use localStorage will required to be cleaned on the first visit of this new page if we go for saving it in cache), not sure what's best

Reply:
You're talking about `firstReceivedPage` and `lastReceivedPage`, right?
https://github.com/algolia/instantsearch.js/pull/4534/commits/83cf619f49fd85a906f4971da382a1135127f447

what if we do not store them, and calculate every time when needed?